tetris.py

```diff
@@ -257,21 +294,17 @@ def full_line_clear(self):
         """
         Clears the line once it's fully populated with blocks
         """
-        y_coordinates = [y for (x, y) in self.get_landed_coords()]
+        y_coordinates = [y for (x, y) in self.coord_extractor()]
         coordinates_counter = collections.Counter(y_coordinates)
         for x_line in range(game_height):
             count = coordinates_counter[x_line]
             if count == game_width:
                 # TODO: root.after() here
-                self.landed_blocks = [
-                    (letter, (a, b))
-                    for (letter, (a, b)) in self.landed_blocks
-                    if b > x_line
-                ] + [
-                    (letter, (a, b + 1))
-                    for (letter, (a, b)) in self.landed_blocks
-                    if b < x_line
-                ]
+                for letter, coord_list in self.landed_blocks.items():
+                    # self.landed_blocks = {letter: [(a, b) for (a, b) in coord_list if b > y_line] + [(a, b+1) for (a, b) in coord_list if b < y_line]} #Why this doesn't work?
```

Comment:
```suggestion
```

The code in this comment does `self.landed_blocks = {letter: something}`. Note that there is no `for` involved, other than inside lists, so this is not a comprehension, just a dict. The created dict has one key, `letter`, with a list as a value. So it replaces `self.landed_blocks` with a new dict that is missing all keys except the one `letter` key.

Reply:
So essentially the comprehension itself happens only inside the list, and not the whole dict.